azurecompute/src/main/java/org/jclouds/azurecompute/compute/AzureComputeServiceAdapter.java

```diff
@@ -88,17 +339,70 @@ public void resumeNode(String id) {
    @Override
    public void suspendNode(String id) {
       // TODO Auto-generated method stub
-
    }
 
    @Override
    public Iterable<Deployment> listNodes() {
-      // TODO Auto-generated method stub
-      return null;
+      Set<Deployment> deployments = FluentIterable.from(api.getCloudServiceApi().list())
+              .transform(new Function<CloudService, Deployment>() {
+                 @Override
+                 public Deployment apply(CloudService cloudService) {
+                    return api.getDeploymentApiForService(cloudService.name()).get(cloudService.name());
+                 }
+              })
+              .filter(notNull())
+              .toSet();
+      return deployments;
    }
 
    @Override public Iterable<Deployment> listNodesByIds(Iterable<String> ids) {
       // TODO Auto-generated method stub
       return null;
    }
+
+   private static class SameLocationAndCreatedStorageServicePredicate implements Predicate<StorageService> {
+      private final String location;
+
+      public SameLocationAndCreatedStorageServicePredicate(String location) {
+         this.location = location;
+      }
+
+      @Override
+      public boolean apply(StorageService input) {
+         return input.storageServiceProperties().location().equals(location) && input.storageServiceProperties().status().equals("Created");
+      }
+   }
+
+   private boolean checkAvailability(String name) {
+      return api.getStorageAccountApi().checkAvailable(subscriptionId, name).result();
+   }
+
+   @VisibleForTesting
+   public static String generateStorageServiceName(String prefix) {
+      String characters = "abcdefghijklmnopqrstuvwxyz";
+      StringBuilder builder = new StringBuilder();
+      builder.append(prefix);
+      int charactersLength = characters.length();
+      for (int i = 0; i < 10; i++) {
+         double index = Math.random() * charactersLength;
+         builder.append(characters.charAt((int) index));
+      }
+      return builder.toString();
+   }
```

Comment:
Move this to a Supplier that can be injected wherever it is needed. This way you have a clean way of generating storage service names. Consider using the `GroupNamingConvention` to generate it properly prefixed, if needed.